test/functional/feature_onlynet.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test -onlynet configuration option.
+"""
+
+from test_framework.basic_server import (
+    BasicServer,
+    tor_control,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+
+class OnlynetTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.extra_args = [
+            ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1'],
+            ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1', '-onion=127.0.0.1:9050'],
+            ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1', '-proxy=127.0.0.1:9050'],
+        ]
```

Comment:
Would it make sense to test with multiple onlynet settings?

```diff
diff --git a/test/functional/feature_onlynet.py b/test/functional/feature_onlynet.py
index 60e4caf33d..c43a541196 100755
--- a/test/functional/feature_onlynet.py
+++ b/test/functional/feature_onlynet.py
@@ -22,6 +22,8 @@ class OnlynetTest(BitcoinTestFramework):
             ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1'],
             ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1', '-onion=127.0.0.1:9050'],
             ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1', '-proxy=127.0.0.1:9050'],
+            ['-onlynet=ipv4', '-onlynet=ipv6', '-torcontrol=127.0.0.1:19051', '-listenonion=1'],
         ]
         self.num_nodes = len(self.extra_args)

@@ -39,7 +41,7 @@ class OnlynetTest(BitcoinTestFramework):
-        for node, expected_reachable in [[0, False], [1, True], [2, True]]:
+        for node, expected_reachable in [[0, False], [1, True], [2, True], [3, False]]:
             self.log.info(f'Verifying node {node} which uses {self.extra_args[node]}')
             assert_equal(self.onion_is_reachable(node), expected_reachable)
```

Reply:
I think it is fine as it is. The point is to have `onlynet=...` but not `onlynet=onion`.